src/test/regress/expected/pg15.out

```diff
@@ -465,6 +465,77 @@ SELECT * FROM numeric_scale_gt_precision WHERE numeric_column=0.027;
           0.027
 (1 row)
 
+-- test repartition joins on tables distributed on numeric types with negative scale
+CREATE TABLE numeric_repartition_first(id int, data int, numeric_column numeric(3,-1));
+CREATE TABLE numeric_repartition_second(id int, data int, numeric_column numeric(3,-1));
+-- populate tables
+INSERT INTO numeric_repartition_first SELECT x, x, x FROM generate_series (100, 115) x;
+INSERT INTO numeric_repartition_second SELECT x, x, x FROM generate_series (100, 115) x;
+-- Run some queries before distributing the tables to see results in vanilla PG
+SELECT count(*)
+FROM numeric_repartition_first f,
+     numeric_repartition_second s
+WHERE f.id = s.numeric_column;
+ count
+---------------------------------------------------------------------
+    15
+(1 row)
+
+SELECT count(*)
+FROM numeric_repartition_first f,
+     numeric_repartition_second s
+WHERE f.numeric_column = s.numeric_column;
+ count
+---------------------------------------------------------------------
+   126
+(1 row)
+
+-- distribute tables and re-run the same queries
+SELECT * FROM create_distributed_table('numeric_repartition_first','id');
+NOTICE:  Copying data from local table...
+NOTICE:  copying the data has completed
+DETAIL:  The local data in the table is no longer visible, but is still on disk.
+HINT:  To remove the local data, run: SELECT truncate_local_data_after_distributing_table($$pg15.numeric_repartition_first$$)
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT * FROM create_distributed_table('numeric_repartition_second','id');
+NOTICE:  Copying data from local table...
+NOTICE:  copying the data has completed
+DETAIL:  The local data in the table is no longer visible, but is still on disk.
+HINT:  To remove the local data, run: SELECT truncate_local_data_after_distributing_table($$pg15.numeric_repartition_second$$)
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SET citus.enable_repartition_joins TO 1;
+SELECT count(*)
+FROM numeric_repartition_first f,
+     numeric_repartition_second s
+WHERE f.id = s.numeric_column;
+ERROR:  complex joins are only supported when all distributed tables are joined on their distribution columns with equal operator
+-- show that the same query works if we use an int column instead of a numeric on the filter clause
+SELECT count(*)
+FROM numeric_repartition_first f,
+     numeric_repartition_second s
+WHERE f.id = s.data;
+ count
+---------------------------------------------------------------------
+    16
+(1 row)
```

Comment:
I do not expect one query to fail and other to succeed. The only difference is on the filter clause. Neither `s.numeric_column` nor `s.data` are distribution columns. The only difference is their types that are `numeric(3, -1)` and `int` respectively.

There should also be some implicit casting going on when using the numeric type on the filter.

Reply:
That error is a known limitation in Citus, it is not supported to have a join on different data types:

```SQL
create table d_int(a int, b int);
select create_distributed_table('d_int', 'a') ;
select create_distributed_table('d_bigint', 'a') ;
select * from d_int JOIN d_bigint USING(a);
ERROR:  complex joins are only supported when all distributed tables are joined on their distribution columns with equal operator
```

See https://github.com/citusdata/citus/issues/2106 (though the error message seems to have changed)

Reply:
How can I properly test shard pruning in repartition joins if we can not distribute by numeric types with negative scale, and we do not allow filters on numeric and another type that is supported for distribution?

By the way I think the error message is a bit misleading here as the problem is with type mismatch.